src/Simulation/Simulators/CommonNativeSimulator/Dump.cs

```diff
@@ -96,5 +92,393 @@ public QSimDumpRegister(CommonNativeSimulator m) : base(m)
                 return Simulator.Dump(location, qubits);
             };
         }
+
+        /// <summary>
+        ///     The convention to be used in labeling computational basis states
+        ///     given their representations as strings of classical bits.
+        /// </summary>
+        [JsonConverter(typeof(StringEnumConverter))]
+        public enum BasisStateLabelingConvention
```

Comment:
[nit] As a minor organizational thing, this file is getting a bit large; since the containing class is `partial` anyway, it may be worth splitting this and `DisplayableState` into a new file.

Reply:
Extracted the `DisplayableState` into a separate file.